File: pkgs/primer_py/primer_py/ejemplo_2.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import Int32


class SubcriberNode(Node):
    def __init__(self):
        super().__init__('subcriber_node')
        self.create_subscription(Int32,'publisher_topic',
                                  self.listener_callback, 10)

        self.get_logger().info("El nodo subcritor esta activo")


    def listener_callback(self, msg):

        self.get_logger().info(f'El mensaje recibido es: "{msg.data}')


def main(args=None):
    rclpy.init(args=args)
    node = SubcriberNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: pkgs/primer_cpp/src/ejemplo_2.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int32.hpp"

class MiSuscriptor : public rclcpp::Node
{
public:
    MiSuscriptor() : Node("suscriptor_cpp")
    {
        // Suscripción al tópico: (Nombre del topic, Calidad de servicio QoS, Callback)
        subscription_ = this->create_subscription<std_msgs::msg::Int32>(
            "publisher_topic",
            10,
            std::bind(&MiSuscriptor::topic_callback, this, std::placeholders::_1)
        );

        RCLCPP_INFO(this->get_logger(), "Nodo suscriptor C++ listo y escuchando.");
    }

private:
    void topic_callback(const std_msgs::msg::Int32::SharedPtr msg) const
    {
        RCLCPP_INFO(this->get_logger(), "Mensaje recibido: %d", msg->data);
    }

    rclcpp::Subscription<std_msgs::msg::Int32>::SharedPtr subscription_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MiSuscriptor>());
    rclcpp::shutdown();
    return 0;
}